torchvision/models/detection/_utils.py

```diff
@@ -375,7 +378,8 @@ def __call__(self, match_quality_matrix: Tensor) -> Tensor:
 
         if self.allow_low_quality_matches:
             if all_matches is None:
-                raise ValueError("all_matches should not be None")
+                torch._assert(False, "all_matches should not be None")
+                return torch.empty((0), dtype=torch.int64)  # not reachable - added to make type checker happy
```

Comment:
So clearly this is one of the things we need to do to please JIT/FX, so it's going to be ugly. Between adding a fake return and using an `if/else` idiom (when possible) I would much prefer the latter. Is the fake return necessary in this case or you can just use an else clause for the method below?

Reply:
Else works I think (i made so many changes I am not sure anymore but I can test again). 
But I think despite both are ugly , the else option would have to ident the entire body which seems to be more distracting than the fake return.

Reply:
Indeed, we are between 2 problematic solutions here. 

Note that this won't be the only place where we have "fake code" to handle JIT (we have fake [casts](https://github.com/pytorch/vision/blob/a0867fac8dd13e436a4f756d3de8b9d6768c391e/torchvision/models/detection/transform.py#L19-L22) and [typing](https://github.com/pytorch/vision/blob/a0867fac8dd13e436a4f756d3de8b9d6768c391e/torchvision/transforms/functional.py#L364)) but we typically use them as last resort. Here a valid idiom exists (using `else`) which is why I think it should be preferred.